pkg/ruler/api.go

```diff
@@ -394,6 +394,19 @@ func (a *API) ListRules(w http.ResponseWriter, req *http.Request) {
 
 	level.Debug(logger).Log("msg", "retrieving rule groups with namespace", "userID", userID, "namespace", namespace)
 	rgs, err := a.store.ListRuleGroupsForUserAndNamespace(req.Context(), userID, namespace)
+	if len(rgs) == 0 && strings.Contains(userID, "|") { //get rules for every tenant to support teant federation
```

Comment:
Thanks @friedrichg, i will check that and try if it chould be implemented using mergeable.
Cortext already support tenant federation, currently it can not support ruler federation,
I also check the mimir ruler.federation functions, i don't think that big value as it will migrate data between different tenant, I feel for the end user, they list all the rules and related information is enough

Reply:
@friedrichg , i studied the mergeable approach, it is helpful, I think it suitable for querying metric, as it query a log metric data parallelly. but for list rules or rulegroups, it will be better to iterate rules for simple reason.
If one day the Ruler could support federation feature(I mean one rule could select from muti tenant data), we can use the mergeable in that case

Reply:
i agree with your opinion and refact the code